### PR TITLE
Add possibility to add a new contact to ContactList

### PR DESCRIPTION
**Pull request by Yana-Kalandarova**

### CODE REVIEW

File: src/components/AddContactPage.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import ContactForm from './ContactForm';
+
+const propTypes = {
+  state: PropTypes.objectOf(PropTypes.shape({
```
Comment (maximshenets) [inline src/components/AddContactPage.js]: Very wrong approach. Never combine different props to generic object, should be two separate props

Comment (Yana-Kalandarova) [reply src/components/AddContactPage.js]: divided to separate props (searchValue and contactList)

File: src/components/ContactForm.js [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+    return (
+      <form className="contact_form" onSubmit={this.handleSubmit}>
+        <div className="contact_form-field-wrap">
+          <label htmlFor="name">Name:</label>
+          <input
+            type="text"
+            placeholder="Name"
+            id="name"
+            name="name"
+            value={this.state.name}
```
Comment (maximshenets) [inline src/components/ContactForm.js]: Please use destructuring at the beginning of the method, you have a lot of `this.state....`

File: src/components/app.js [hunk 3]
```diff
@@ -28,19 +28,34 @@ class App extends Component {
...
 
     return (
-      <div className="view_wrap">
-        <h1 className="view_title">Contact List</h1>
-        <SearchBar onSearch={this.handleSearch} search={search} />
-        <ContactList contactList={filterList(contactList, search)} onDeleteItem={this.handleDeleteItem} />
-      </div>
+      <BrowserRouter>
+        <Route exact path="/" render={
+          () => <ContactListPage searchValue={this.state.searchValue} contactList={this.state.contactList} onDeleteContact={this.handleDeleteContact} onSearchContact={this.handleSearchContact} />
```
Comment (maximshenets) [inline src/components/app.js]: Why you use `this.state` here, you destructured this values above

File: src/components/ContactListPage.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+import { Link } from 'react-router-dom';
+import PropTypes from 'prop-types';
+import { filterList } from '../utils/index';
+
+import SearchBar from './SearchBar';
+import ContactList from './ContactList';
+
+const propTypes = {
+  searchValue: PropTypes.string.isRequired,
+  contactList: PropTypes.arrayOf(PropTypes.objectOf(PropTypes.shape({
```
Comment (maximshenets) [inline src/components/ContactListPage.js]: I got some errors in console about prop types. This cuz you use `objectOf(shape(....))`. objectOf should be removed cuz shape it is an object

File: src/components/ContactForm.js (hunk 2 above)
Comment (Yana-Kalandarova) [reply src/components/ContactForm.js]: Updated.

File: src/components/app.js (hunk 3 above)
Comment (Yana-Kalandarova) [reply src/components/app.js]: Updated, replaced with destructed value.

File: src/components/ContactListPage.js (hunk 4 above)
Comment (Yana-Kalandarova) [reply src/components/ContactListPage.js]: Removed. Thanks!

**Pull request merged**